### ISSUE TITLE
Different results in /search/things and /search/things/count requests

### ISSUE THREAD

**Issue by w4tsn** [eclipse/ditto#1434]
Ditto Version: 2.4

I'm observing some strange behavior of the search API: doing a request with or without filters to the /count endpoint yields a different number than the result of the /search query (using the same request, only different endpoints).

**Example 1:**

I have a realm with roughly 2200 things in about 150 policies. Maybe it's related to permissions?

I see that although the count returns around 2200, the first page of the /search request with a `size(200)` option only returns 191 items. The next page then has 200 items. If I inverse the sort order to descending, the first page has 200 results.

**Example 2:**

In this example what's notable is that inversing the sort order does not change the number of results in the first page. I'm not sure yet why this is.

```
GET /search/things/count
=> 1074
GET /search/things
=> 25
GET /search/things&option=size(25)
=> 25
GET /search/things&option=size(150)
=> 145
GET /search/things&option=size(200)
=> 191
GET /search/things&option=size(200)&sort(-)
=> 191
```
**More testing:**

It seems that the /search/things endpoint gathers some things it does not return. It computes a bucket of 200 things but skips certain one of them. I'm currently investigating what those things are and which differences they have to the ones that are included in the search result.

The deprecated `limit` option behaves the same as `size,cursor`

**Reproduction:**

I was not yet able to reproduce this on the sandbox instance. So unfortunately I do not have a reproducer just yet. Any help in debugging / coming up with a repro is appreciated.

**Comment by thjaeckle** [comment]
We also noticed that for Ditto 2.x versions there could be situations where inconsistencies in the search can occur.
That was due to the nature of the chosen "attribute pattern" (together with only partially updating the index by doing array "splices", etc.) we applied in the MongoDB for building the generic search index.
Among other performance issues of the "old search" we fixed this problem with the new "wildcard index" approach in #1374

So this should be fixed for Ditto 3.0.0 as well.
Backporting the fix to 2.4 is unfortunately not possible as the complete DB schema of the search changed - that's also one of the reasons for a new major version of Ditto.

If this helps, we can soon offer a 3.0.0-M1 version - the 3.0.0 release will probably done in a 1-2 months timeframe.
